src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+    );
+    const diffDiff = new p5.Vector(
+      cam0.eyeX - cam1.eyeX - cam0.centerX + cam1.centerX,
+      cam0.eyeY - cam1.eyeY - cam0.centerY + cam1.centerY,
+      cam0.eyeZ - cam1.eyeZ - cam0.centerZ + cam1.centerZ
+    );
+    // Suppose there are two line segments. Consider the distance between the points
+    // above them as if they were taken in the same ratio. This calculation figures out
+    // a ratio that minimizes this.
+    // Each line segment is, a line segment connecting the viewpoint and the center
+    // for each camera.
+    const divider = diffDiff.magSq();
+    let ratio = 1; // default.
+    if (divider > 0.000001){
+      ratio = p5.Vector.dot(eyeDiff, diffDiff) / divider;
+      ratio = Math.max(0, Math.min(ratio, 1));
+    }
+
+    // Take the appropriate proportions and work out the points
+    // that are between the new viewpoint and the new center position.
+    const medium0X = (1-ratio) * cam0.eyeX + ratio * cam0.centerX;
+    const medium0Y = (1-ratio) * cam0.eyeY + ratio * cam0.centerY;
+    const medium0Z = (1-ratio) * cam0.eyeZ + ratio * cam0.centerZ;
+
+    const medium1X = (1-ratio) * cam1.eyeX + ratio * cam1.centerX;
+    const medium1Y = (1-ratio) * cam1.eyeY + ratio * cam1.centerY;
+    const medium1Z = (1-ratio) * cam1.eyeZ + ratio * cam1.centerZ;
+
+    const lerpedMediumX = medium0X * (1-amt) + medium1X * amt;
+    const lerpedMediumY = medium0Y * (1-amt) + medium1Y * amt;
+    const lerpedMediumZ = medium0Z * (1-amt) + medium1Z * amt;
+
+    // Prepare each of rotation matrix from their camera matrix
+    const mat0 = cam0.cameraMatrix.mat4;
+    const mat1 = cam1.cameraMatrix.mat4;
+    const m0 = [
+      mat0[0], mat0[1], mat0[2],
+      mat0[4], mat0[5], mat0[6],
+      mat0[8], mat0[9], mat0[10]
+    ];
+    const m1 = [
+      mat1[0], mat1[1], mat1[2],
+      mat1[4], mat1[5], mat1[6],
+      mat1[8], mat1[9], mat1[10]
+    ];
+    // Create the inverse matrix of mat0 by transposing mat0,
+    // and multiply it to mat1 from the right.
+    // This matrix represents the difference between the two.
+    const m = [
+      m1[0] * m0[0] + m1[1] * m0[1] + m1[2] * m0[2],
+      m1[0] * m0[3] + m1[1] * m0[4] + m1[2] * m0[5],
+      m1[0] * m0[6] + m1[1] * m0[7] + m1[2] * m0[8],
+
+      m1[3] * m0[0] + m1[4] * m0[1] + m1[5] * m0[2],
+      m1[3] * m0[3] + m1[4] * m0[4] + m1[5] * m0[5],
+      m1[3] * m0[6] + m1[4] * m0[7] + m1[5] * m0[8],
+
+      m1[6] * m0[0] + m1[7] * m0[1] + m1[8] * m0[2],
+      m1[6] * m0[3] + m1[7] * m0[4] + m1[8] * m0[5],
+      m1[6] * m0[6] + m1[7] * m0[7] + m1[8] * m0[8]
+    ];
+
+    // Calculate the trace and from it the cos value of the angle.
+    // An orthogonal matrix is just an orthonormal basis. If this is not the identity
+    // matrix, it is a centered orthonormal basis plus some angle of rotation about
+    // some axis. That's the angle. Letting this be theta, trace becomes 1+2cos(theta).
```

Comment:
Since this might not be obvious to readers, maybe we can drop in a link to this: https://en.wikipedia.org/wiki/Rotation_matrix#Determining_the_angle

Reply:
I learned for the first time that there is a website that has a specific shape of the procession.
Ok, I'll add it.